aspnetcore/client-side/import-export-interop.md

```diff
@@ -0,0 +1,394 @@
+| `Action<T1, T2>`            | `Function`   |            |                   |                        | |
+| `Action<T1, T2, T3>`        | `Function`   |            |                   |                        | |
+| `Func<TResult>`             | `Function`   |            |                   |                        | |
+| `Func<T1, TResult>`         | `Function`   |            |                   |                        | |
+| `Func<T1, T2, TResult>`     | `Function`   |            |                   |                        | |
+| `Func<T1, T2, T3, TResult>` | `Function`   |            |                   |                        | |
+
+The following conditions apply to type mapping and marshalled values:
+
+* The :::no-loc text="Array of"::: column indicates if the .NET type can be marshalled as a JS [`Array`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array). Example: C# `int[]` (`Int32`) mapped to JS `Array` of `Number`s.
+* When passing a JS value to C# with a value of the wrong type, the framework throws an exception in most cases. The framework doesn't perform compile-time type checking in JS.
+* `JSObject`, `Exception`, `Task` and `ArraySegment` create `GCHandle` and a proxy. You can trigger disposal in developer code or allow [.NET garbage collection (GC)](/dotnet/standard/garbage-collection/) to dispose of the objects later. These types carry significant performance overhead.
+* `MemoryView`
+  * `MemoryView` is a .NET type, not a native JS type. It's API is described in the [`dotnet.d.ts` file (`dotnet/runtime` GitHub repository)&dagger;](https://github.com/dotnet/runtime/blob/main/src/mono/wasm/runtime/dotnet.d.ts).
+  * Bytes aren't copied during marshalling.
+  * `MemoryView` created for a `Span` is only valid for the duration of the interop call.
+  * `MemoryView` created for an `ArraySegment` survives after the interop call and is useful for sharing a buffer.
+  * `MemoryView` doesn't have an analogous JS type, so marshalling a JS object to a .NET `MemoryView` isn't possible.
+* It's not possible to export a .NET method that returns a `Span`. The `Span` is allocated on the call stack and has GC implications. When calling from JS to .NET, there's no C# stack after the call.
+* For an exported method that returns an `ArraySegment`, calling `dispose()` in `try-finally` block disposes the proxy and unpins the underlying C# byte array. We recommend calling `dispose()` on the object in developer JS code. If developer code doesn't dispose of the object, the JS GC eventually disposes the object. You can also marshal a byte array (`byte[]`) instead of an `ArraySegment`, which copies the bytes.
+
+&dagger;The link to the `dotnet.d.ts` file in the .NET reference source loads the repository's default branch (`main`), which represents the current development for the next release of .NET. To select a tag for a specific release, use the **Switch branches or tags** dropdown list.
+
```

Comment:
Since MemoryView isn't mentioned in dotnet.d.ts, it doesn't look like this note is actually needed.

```suggestion
```